tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/SetInclusionRuleWithArrays.scala

```diff
@@ -0,0 +1,79 @@
+package at.forsyte.apalache.tla.bmcmt.rules
+
+import at.forsyte.apalache.tla.bmcmt.implicitConversions.Crossable
+import at.forsyte.apalache.tla.bmcmt.rewriter.ConstSimplifierForSmt
+import at.forsyte.apalache.tla.bmcmt.types.{BoolT, FinSetT, PowSetT}
+import at.forsyte.apalache.tla.bmcmt.{ArenaCell, RewriterException, SymbState, SymbStateRewriter}
+import at.forsyte.apalache.tla.lir.{OperEx, TlaEx}
+import at.forsyte.apalache.tla.lir.convenience._
+import at.forsyte.apalache.tla.lir.UntypedPredefs._
+import at.forsyte.apalache.tla.lir.oper.TlaSetOper
+
+class SetInclusionRuleWithArrays(rewriter: SymbStateRewriter) extends SetInclusionRule(rewriter) {
+  private val simplifier = new ConstSimplifierForSmt
+
+  override def apply(state: SymbState): SymbState = {
+    state.ex match {
+      case OperEx(TlaSetOper.subseteq, left, right) =>
+        val leftState = rewriter.rewriteUntilDone(state.setRex(left))
+        val leftCell = leftState.arena.findCellByNameEx(leftState.ex)
+        val rightState = rewriter.rewriteUntilDone(leftState.setRex(right))
+        val rightCell = rightState.arena.findCellByNameEx(rightState.ex)
+        (leftCell.cellType, rightCell.cellType) match {
+          case (FinSetT(_), FinSetT(_)) =>
+            subset(rightState, leftCell, rightCell, false)
+
+          case (FinSetT(FinSetT(t1)), PowSetT(FinSetT(t2))) =>
+            if (t1 != t2) {
+              throw new RewriterException(
+                  "Unexpected set types: %s and %s in %s"
+                    .format(t1, t2, state.ex), state.ex)
+            } else {
+              subset(rightState, leftCell, rightCell, true)
+            }
+
+          case _ =>
```

Comment:
This is equal to the original `SetInclusionRule` rule. I didn't think much about it since an exception is already thrown there. If we make a change it should be in both encodings I think. Maybe in a separate PR?

Reply:
@konnov thoughts?

Reply:
There is `case _ =>` that throws an exception below. So it will be reported as an error. Perhaps, we should throw a better exception there that indicates that this case is not supported.

Reply:
Yeah, but shouldn't 
```
A == {1,2}
B == SUBSET A \subseteq { x \in SUBSET A: x < 5 } \* <=> SUBSET A \subseteq SUBSET A <=> TRUE
```
translate into exactly the case where the LHS is a `PowSetT` and the RHS is a standard `FinSetT` because of the filter?
You can't throw on this example.

Reply:
I am not sure. You can try this example and see what happens. Maybe the left-hand side will get expanded. If it does not, we should fix `ExpansionMarker` to do that.

Reply:
See #1103